packages/expo-file-system/ios/Next/FileSystemDirectory.swift

```diff
@@ -2,9 +2,23 @@ import Foundation
 import ExpoModulesCore
 
 internal final class FileSystemDirectory: FileSystemPath {
+  override init(url: URL) {
+    if url.hasDirectoryPath {
+      super.init(url: url)
+      return
+    }
+    super.init(url: url.deletingLastPathComponent().appendingPathComponent(url.lastPathComponent, isDirectory: true))
```

Comment:
Is this to ensure the url ends with `/`? Can we do this in two lines?
```suggestion
    let directoryUrl = url.deletingLastPathComponent().appendingPathComponent(url.lastPathComponent, isDirectory: true)
    super.init(url: directoryUrl)
```